src/app/Library/CrudPanel/Traits/Update.php

```diff
@@ -85,22 +86,76 @@ public function getUpdateFields($id = false)
      */
     private function getModelAttributeValue($model, $field)
     {
-        if (isset($field['entity'])) {
-            $relational_entity = $this->parseRelationFieldNamesFromHtml([$field])[0]['name'];
+        if (isset($field['entity']) && $field['entity'] !== false) {
+            $relational_entity = $this->getOnlyRelationEntity($field);
 
             $relation_array = explode('.', $relational_entity);
 
-            $relatedModel = $relatedModel = array_reduce(array_splice($relation_array, 0, -1), function ($obj, $method) {
+            $related_model = array_reduce(array_splice($relation_array, 0, -1), function ($obj, $method) {
+                $method = Str::endsWith($method, '_id') ? Str::replaceLast('_id', '', $method) : $method;
+
                 return $obj->{$method} ? $obj->{$method} : $obj;
             }, $model);
 
-            $relationMethod = Arr::last($relation_array);
-
-            if (method_exists($relatedModel, $relationMethod) && $relatedModel->{$relationMethod}() instanceof HasOne) {
-                return $relatedModel->{$relationMethod}->{Arr::last(explode('.', $relational_entity))};
-            } else {
-                return $relatedModel->{$relationMethod};
+            $relation_method = Str::afterLast($relational_entity, '.');
```

Comment:
Don't we use this bit of logic anywhere else? This splits the dot notation into an array, to get the `$related_model` and `$relation_method`, correct? 

If so, this is a stand-alone bit of logic, with clear input and clear output. Moving it to a separate method would improve readability, testability, reusability. I'm not a fan of creating new functions before they need to be reused, but in this case I think this section needs comments for beginning&ending, to say "here we start calculating that", then "here we end calculating that". And what better way to say start/end, than to... not say it at all, and move all that logic to a private method? 😀 It's a bit awkward because the function returns two things, I get it, but I've seen people do this lately in Laravel so it's become standard practice.

```php
public static function getModelAndMethodFromDotNotation($string) 
{
    // ... same logic
    return [$related_model, $relation_method];
}
```

Then we'd just call this method inside `getModelAttributeValueFromRelationship`:

```php
$relational_entity = $this->getOnlyRelationEntity($field);
[$related_model, $relation_method] = static::getModelAndMethodFromDotNotation($relational_entity);
// now we can use $related_model & $relation_method inside the next switch statement
```

Any chance we can do this @pxpm ?